e2e/validate/validation.go

```diff
@@ -181,3 +160,65 @@ func clustersInClusterSet(env *types.Env, config *types.Config, log *zap.Sugared
 
 	return nil
 }
+
+// detectDistro detects the cluster distribution. If distro is set in config, it uses the user
+// provided distro. Otherwise, it determines the distro, sets config.Distro and config.Namespaces.
+// Returns an error if distro detection fails or if clusters have inconsistent distributions.
+func detectDistro(env *types.Env, cfg *types.Config, log *zap.SugaredLogger) error {
+	if cfg.Distro != "" {
+		log.Infof("Using user selected kubernetes distribution: %q", cfg.Distro)
+
+		return nil
+	}
+
+	var detectedDistro string
+
+	clusters := []*types.Cluster{&env.Hub, &env.C1, &env.C2}
+	for _, cluster := range clusters {
+		distro, err := probeDistro(cluster)
+		if err != nil {
+			return fmt.Errorf("failed to determine distro for cluster %q: %w", cluster.Name, err)
+		}
+
+		if detectedDistro == "" {
+			detectedDistro = distro
+		} else if detectedDistro != distro {
+			return fmt.Errorf("clusters have inconsistent distributions, cluster %q has distro %q, expected %q",
+				cluster.Name, distro, detectedDistro)
+		}
+	}
+
+	cfg.Distro = detectedDistro
+	if detectedDistro == config.DistroOcp {
+		cfg.Namespaces = config.OcpNamespaces
+	} else {
+		cfg.Namespaces = config.K8sNamespaces
+	}
+
+	log.Infof("Detected and set kubernetes distribution: %q", cfg.Distro)
```

Comment:
The log is correct but we don't log "set" in any other log. Let' simplify to:

    Detected kubernetes distribution: "ocp"